folio-spring-i18n/src/main/java/org/folio/spring/service/TranslationService.java

```diff
@@ -0,0 +1,347 @@
+package org.folio.spring.service;
+
+import com.ibm.icu.text.ListFormatter;
+import jakarta.annotation.Nonnull;
+import jakarta.annotation.Nullable;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+import org.folio.spring.config.TranslationConfiguration;
+import org.folio.spring.model.TranslationFile;
+import org.folio.spring.model.TranslationFile.LanguageRegionPair;
+import org.folio.spring.model.TranslationMap;
+import org.folio.spring.model.TranslationMatchQuality;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.io.Resource;
+import org.springframework.core.io.support.ResourcePatternResolver;
+import org.springframework.stereotype.Service;
+import org.springframework.web.context.request.RequestContextHolder;
+import org.springframework.web.context.request.ServletRequestAttributes;
+
+/**
+ * A {@link Service Service} to manage translations as a whole.  In reality, the most important
+ * method is simply the format method which will do all of the heavy lifting as needed.  This
+ * service will keep track of translation files and locales, caching as possible, in order to
+ * serve the user the best possible translation.
+ */
+@Log4j2
+@Service
+@RequiredArgsConstructor(onConstructor_ = @Autowired)
+public class TranslationService {
+
+  /**
+   * The location of all translations in the classpath. Allows for format matching UI modules:
+   *  /translations/{module-name}/{locale}.json
+   */
+  public static final String TRANSLATIONS_CLASSPATH = "classpath:%s*/*.json";
+
+  /**
+   * A map from language code -&gt; country -&gt; JSON resource containing {key -&gt; ICU format string}.
```

Comment:
Maybe something like this, to make the docs here more readable when looking the source code
```suggestion
   * A map from language code to a map from country to the JSON resource containing {key to ICU format string mappings}.
```

Reply:
Sure, but it also renders nicely when hovering over too!